vpd-manager/src/worker.cpp

```diff
@@ -144,22 +145,38 @@ void Worker::performInitialSetup()
         // some reason at system power on.
         return;
     }
-    catch (const std::exception& ex)
+    catch (const std::exception& l_ex)
     {
-        if (typeid(ex) == std::type_index(typeid(DataException)))
+        if (typeid(l_ex) == std::type_index(typeid(DataException)))
         {
-            // TODO:Catch logic to be implemented once PEL code goes in.
+            EventLogger::createAsyncPel(types::ErrorType::InvalidVpdMessage,
+                                        types::SeverityType::Informational,
+                                        __FILE__, __FUNCTION__, 0, l_ex.what(),
+                                        std::nullopt, std::nullopt,
+                                        std::nullopt, std::nullopt);
         }
-        else if (typeid(ex) == std::type_index(typeid(EccException)))
+        else if (typeid(l_ex) == std::type_index(typeid(EccException)))
         {
-            // TODO:Catch logic to be implemented once PEL code goes in.
+            EventLogger::createAsyncPel(types::ErrorType::EccCheckFailed,
+                                        types::SeverityType::Informational,
+                                        __FILE__, __FUNCTION__, 0, l_ex.what(),
+                                        std::nullopt, std::nullopt,
+                                        std::nullopt, std::nullopt);
         }
-        else if (typeid(ex) == std::type_index(typeid(JsonException)))
+        else if (typeid(l_ex) == std::type_index(typeid(JsonException)))
         {
-            // TODO:Catch logic to be implemented once PEL code goes in.
+            EventLogger::createAsyncPel(types::ErrorType::JsonFailure,
+                                        types::SeverityType::Informational,
+                                        __FILE__, __FUNCTION__, 0, l_ex.what(),
+                                        std::nullopt, std::nullopt,
+                                        std::nullopt, std::nullopt);
         }
 
-        logging::logMessage(ex.what());
+        EventLogger::createAsyncPel(
+            types::ErrorType::InvalidEeprom, types::SeverityType::Informational,
+            __FILE__, __FUNCTION__, 0, l_ex.what(), std::nullopt, std::nullopt,
+            std::nullopt, std::nullopt);
+        logging::logMessage(l_ex.what());
         throw;
```

Comment:
Since we have already logged PEL here in this flow. We need not throw again.
Let the manager service come up with error.

Reply:
Done